client/driver/docker.go

```diff
@@ -333,6 +333,19 @@ func (d *DockerDriver) Open(ctx *ExecContext, handleID string) (DriverHandle, er
 	return h, nil
 }
 
+func (d *DockerDriver) dockerClient() (*docker.Client, error) {
+	dockerEndpoint := d.config.Read("docker.endpoint")
+	var client *docker.Client
+	var err error
+	if dockerEndpoint == "" {
+		client, err = docker.NewClientFromEnv()
+	} else {
+		client, err = docker.NewClient(dockerEndpoint)
+	}
+
+	return client, err
```

Comment:
Instead of declaring `client` and `err` up front, you could just return from inside of the `if` blocks.